### PR TITLE
kclient: refactor for easier usage

### PR DESCRIPTION
**Pull request by howardjohn**
This is just refactoring. Mostly, its to make it easier to use for tests when we want either direct reads (Get and List w/o informer) or to just generate a writer-only client w/o and informer.

Tests are updated to use these.

### CODE REVIEW

File: pkg/kube/kclient/interfaces.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+type ReadWriter[T controllers.Object] interface {
+	Reader[T]
+	Writer[T]
+}
+
+// Client wraps a Kubernetes client providing cached read access and direct write access.
+type Client[T controllers.Object] interface {
+	Reader[T]
+	Writer[T]
+	Informer[T]
```
Comment (hzxuzhonghu) [inline pkg/kube/kclient/interfaces.go:73]: Informer has included Reader

File: pkg/kube/kclient/interfaces.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+// This is based on informers, so most of the same caveats to informers apply here.
+type Reader[T controllers.Object] interface {
+	// Get looks up an object by name and namespace. If it does not exist, nil is returned
+	Get(name, namespace string) T
+	// List looks up an object by namespace and labels.
+	// Use metav1.NamespaceAll and klabels.Everything() to select everything.
+	List(namespace string, selector klabels.Selector) []T
+}
+
+type Informer[T controllers.Object] interface {
```
Comment (hzxuzhonghu) [inline pkg/kube/kclient/interfaces.go:36]: It is not clear what's the diff with kubernetes `SharedInformer`

Comment (howardjohn) [reply pkg/kube/kclient/interfaces.go:36]: If you look at the methods you will see its quite different, the only function in common is AddEventHandler and HasSynced, and both of those functions have different implementations for this.

`Informer` is just the only interface `kclient` implemented, it is a Reader (Get + List) + informer attributes (handler, synced, etc)

**Pull request merged**